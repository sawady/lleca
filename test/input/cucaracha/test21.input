

fun f(a : Int) {
  a := a - 1
  putNum(
    a + ( a + ( a + ( a - ( a + ( a + ( a + ( a - ( a + ( a - (
    a + ( a - ( a + ( a + ( a + ( a - ( a + ( a - ( a - ( a + (
    a + ( a - ( a - ( a - ( a + ( a - ( a + ( a - ( a + ( a - (
    a - ( a + ( a - ( a + ( a + ( a - ( a + ( a + ( a - ( a - (
    a - a))))))))))))))))))))))))))))))))))))))))
  )
  putChar(10)
}

fun main() {
  f(1)
  f(2)
  f(3)
  f(4)
  f(5)
  f(6)
  f(7)
  f(8)
  f(9)
  f(10)
}
