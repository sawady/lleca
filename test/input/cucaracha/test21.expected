Cons(
  Function(
    f,
    Cons(
      Parameter(
        a,
        Int
      ),
      Nil
    ),
    Unit,
    Cons(
      StmtAssign(
        a,
        ExprSub(
          ExprVar(
            a
          ),
          ExprConstNum(
            1
          )
        )
      ),
      Cons(
        StmtCall(
          putNum,
          Cons(
            ExprAdd(
              ExprVar(
                a
              ),
              ExprAdd(
                ExprVar(
                  a
                ),
                ExprAdd(
                  ExprVar(
                    a
                  ),
                  ExprSub(
                    ExprVar(
                      a
                    ),
                    ExprAdd(
                      ExprVar(
                        a
                      ),
                      ExprAdd(
                        ExprVar(
                          a
                        ),
                        ExprAdd(
                          ExprVar(
                            a
                          ),
                          ExprSub(
                            ExprVar(
                              a
                            ),
                            ExprAdd(
                              ExprVar(
                                a
                              ),
                              ExprSub(
                                ExprVar(
                                  a
                                ),
                                ExprAdd(
                                  ExprVar(
                                    a
                                  ),
                                  ExprSub(
                                    ExprVar(
                                      a
                                    ),
                                    ExprAdd(
                                      ExprVar(
                                        a
                                      ),
                                      ExprAdd(
                                        ExprVar(
                                          a
                                        ),
                                        ExprAdd(
                                          ExprVar(
                                            a
                                          ),
                                          ExprSub(
                                            ExprVar(
                                              a
                                            ),
                                            ExprAdd(
                                              ExprVar(
                                                a
                                              ),
                                              ExprSub(
                                                ExprVar(
                                                  a
                                                ),
                                                ExprSub(
                                                  ExprVar(
                                                    a
                                                  ),
                                                  ExprAdd(
                                                    ExprVar(
                                                      a
                                                    ),
                                                    ExprAdd(
                                                      ExprVar(
                                                        a
                                                      ),
                                                      ExprSub(
                                                        ExprVar(
                                                          a
                                                        ),
                                                        ExprSub(
                                                          ExprVar(
                                                            a
                                                          ),
                                                          ExprSub(
                                                            ExprVar(
                                                              a
                                                            ),
                                                            ExprAdd(
                                                              ExprVar(
                                                                a
                                                              ),
                                                              ExprSub(
                                                                ExprVar(
                                                                  a
                                                                ),
                                                                ExprAdd(
                                                                  ExprVar(
                                                                    a
                                                                  ),
                                                                  ExprSub(
                                                                    ExprVar(
                                                                      a
                                                                    ),
                                                                    ExprAdd(
                                                                      ExprVar(
                                                                        a
                                                                      ),
                                                                      ExprSub(
                                                                        ExprVar(
                                                                          a
                                                                        ),
                                                                        ExprSub(
                                                                          ExprVar(
                                                                            a
                                                                          ),
                                                                          ExprAdd(
                                                                            ExprVar(
                                                                              a
                                                                            ),
                                                                            ExprSub(
                                                                              ExprVar(
                                                                                a
                                                                              ),
                                                                              ExprAdd(
                                                                                ExprVar(
                                                                                  a
                                                                                ),
                                                                                ExprAdd(
                                                                                  ExprVar(
                                                                                    a
                                                                                  ),
                                                                                  ExprSub(
                                                                                    ExprVar(
                                                                                      a
                                                                                    ),
                                                                                    ExprAdd(
                                                                                      ExprVar(
                                                                                        a
                                                                                      ),
                                                                                      ExprAdd(
                                                                                        ExprVar(
                                                                                          a
                                                                                        ),
                                                                                        ExprSub(
                                                                                          ExprVar(
                                                                                            a
                                                                                          ),
                                                                                          ExprSub(
                                                                                            ExprVar(
                                                                                              a
                                                                                            ),
                                                                                            ExprSub(
                                                                                              ExprVar(
                                                                                                a
                                                                                              ),
                                                                                              ExprVar(
                                                                                                a
                                                                                              )
                                                                                            )
                                                                                          )
                                                                                        )
                                                                                      )
                                                                                    )
                                                                                  )
                                                                                )
                                                                              )
                                                                            )
                                                                          )
                                                                        )
                                                                      )
                                                                    )
                                                                  )
                                                                )
                                                              )
                                                            )
                                                          )
                                                        )
                                                      )
                                                    )
                                                  )
                                                )
                                              )
                                            )
                                          )
                                        )
                                      )
                                    )
                                  )
                                )
                              )
                            )
                          )
                        )
                      )
                    )
                  )
                )
              )
            ),
            Nil
          )
        ),
        Cons(
          StmtCall(
            putChar,
            Cons(
              ExprConstNum(
                10
              ),
              Nil
            )
          ),
          Nil
        )
      )
    )
  ),
  Cons(
    Function(
      main,
      Nil,
      Unit,
      Cons(
        StmtCall(
          f,
          Cons(
            ExprConstNum(
              1
            ),
            Nil
          )
        ),
        Cons(
          StmtCall(
            f,
            Cons(
              ExprConstNum(
                2
              ),
              Nil
            )
          ),
          Cons(
            StmtCall(
              f,
              Cons(
                ExprConstNum(
                  3
                ),
                Nil
              )
            ),
            Cons(
              StmtCall(
                f,
                Cons(
                  ExprConstNum(
                    4
                  ),
                  Nil
                )
              ),
              Cons(
                StmtCall(
                  f,
                  Cons(
                    ExprConstNum(
                      5
                    ),
                    Nil
                  )
                ),
                Cons(
                  StmtCall(
                    f,
                    Cons(
                      ExprConstNum(
                        6
                      ),
                      Nil
                    )
                  ),
                  Cons(
                    StmtCall(
                      f,
                      Cons(
                        ExprConstNum(
                          7
                        ),
                        Nil
                      )
                    ),
                    Cons(
                      StmtCall(
                        f,
                        Cons(
                          ExprConstNum(
                            8
                          ),
                          Nil
                        )
                      ),
                      Cons(
                        StmtCall(
                          f,
                          Cons(
                            ExprConstNum(
                              9
                            ),
                            Nil
                          )
                        ),
                        Cons(
                          StmtCall(
                            f,
                            Cons(
                              ExprConstNum(
                                10
                              ),
                              Nil
                            )
                          ),
                          Nil
                        )
                      )
                    )
                  )
                )
              )
            )
          )
        )
      )
    ),
    Nil
  )
)
